ztp/policygenerator/policyGen/policyBuilder_test.go

```diff
@@ -0,0 +1,202 @@
+package policyGen
+
+import (
+	utils "github.com/openshift-kni/cnf-features-deploy/ztp/policygenerator/utils"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v3"
+)
+
+func extractCRsFromPolicies(t *testing.T, policies map[string]interface{}) []utils.ObjectTemplates {
+	// The policies map contains entries such as:
+	// test1/test1-ptp-sub-policy This is the one we want
+	// test1/test1-placementrules
+	// test1/test1-placementbinding
+	assert.Equal(t, len(policies), 3, "Expect a single policy with placement rule/binding for testing")
+	for key, value := range policies {
+		if strings.Contains(key, "placement") {
+			continue
+		}
+		// This is the configuration policy
+		policy := value.(utils.AcmPolicy)
+		// This is the policy-templates array
+		assert.Equal(t, len(policy.Spec.PolicyTemplates), 1)
+		// Extract the object-template from the object-definitions. The
+		// object-template contains the actual CR
+		objects := policy.Spec.PolicyTemplates[0].ObjDef.Spec.ObjectTemplates
+		// Return the first (and only) non-placement entry
+		return objects
+	}
+	return nil
+}
+
+// Test cases for override of complianceType for Namespace kinds. Namespace as the first object here.
+func TestComplianceTypeDefault(t *testing.T) {
+	input := `
+apiVersion: ran.openshift.io/v1
+kind: PolicyGenTemplate
+metadata:
+  name: "test1"
+  namespace: "test1"
+spec:
+  bindingRules:
+    justfortest: "true"
+  sourceFiles:
+    # Create operators policies that will be installed in all clusters
+    - fileName: PtpSubscriptionNS.yaml
+      policyName: "ptp-sub-policy"
+    - fileName: PtpSubscription.yaml
+      policyName: "ptp-sub-policy"
+    - fileName: PtpSubscriptionOperGroup.yaml
+      policyName: "ptp-sub-policy"
+`
+	// Read in the test PGT
+	pgt := utils.PolicyGenTemplate{}
+	_ = yaml.Unmarshal([]byte(input), &pgt)
+
+	// Set up the files handler to pick up local source-crs and skip any output
+	fHandler := utils.NewFilesHandler("../../source-crs", "/dev/null", "/dev/null")
```

Comment:
can we use the testData/ files instead ? or add new files to the testData/ if need it.